### PR TITLE
Story/hfi page header mods/1650

### PR DESCRIPTION
**Pull request by andrea-williams**
# Test Links:
[Percentile Calculator](https://wps-pr-1703.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1703.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1703.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1703.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1703.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1703.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1703.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+              }
+              onChange={props.selectNewFireCentre}
+            />
+          </FormControl>
+        </Grid>
+        <Grid item md={4}>
+          {/* empty Grid item for spacing */}
+        </Grid>
+
+        <Grid item md={2} justifyContent="flex-end" alignItems="center">
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx:113]: I think you can put this in it's own flex container with `flex-direction: row-reverse` so it starts at the rightmost space of the container. Then you can delete the empty grid items above.

File: web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    },
+    '&:focus .MuiOutlinedInput-notchedOutline': {
+      borderColor: 'white'
+    }
+  }
+}))
+
+interface Props {
+  padding?: string
+  formControlClass: string
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx:67]: `HFIPageSubHeader` can just import the `formControl` class like other components do so you can save passing another prop.

**Review by conbrad — CHANGES_REQUESTED** [review]
Looking good

File: web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    paddingRight: '1rem',
+    fontSize: '1.3rem',
+    background: theme.palette.primary.light,
+    color: theme.palette.primary.contrastText,
+    alignContent: 'center'
+  },
+  gridContainer: {
+    height: '85%'
+  },
+  positionStyler: {
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx:31]: I think this class is unused and can be deleted

File: web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  },
+  aboutButtonText: {
+    color: 'white',
+    textDecoration: 'underline',
+    fontWeight: 'bold',
+    justifyContent: 'flex-end'
+  },
+  dateOfInterestPicker: {
+    marginLeft: '7px',
+    marginRight: 30,
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx:46]: Can we remove the margin settings and just keep the `minWidth`?

File: web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+      <Grid
+        container
+        spacing={0}
+        alignItems="center"
+        direction="row"
+        className={classes.gridContainer}
+      >
+        <Grid item md={3} lg={2}>
+          <FormControl
+            className={`${classes.formControl} ${classes.dateOfInterestPicker}`}
```
Comment (conbrad) [inline web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx:105]: This is my bad, the existing `formControl` has the `minWidth` we want, but also adds margin spacing. I've toggled that spacing off in devtools and the components line up in the centre of the sub header component.

I think just creating our own class with `minWidth: 210` makes sense here.

**Review by conbrad — APPROVED** [review]
Looks great, just some minor changes requested

File: web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx (hunk 3 above)
Comment (andrea-williams) [reply web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx]: Good eye. Deleted

File: web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx (hunk 5 above)
Comment (andrea-williams) [reply web/src/features/hfiCalculator/components/HFIPageSubHeader.tsx]: Done.

**Review by Sybrand — APPROVED** [review]

**Pull request merged**